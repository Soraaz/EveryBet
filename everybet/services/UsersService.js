"use strict";

const HTTPClient = require("./HTTPClient");
const Common = require('./Common');

class UsersService {

    // POST

    static createNewUser(login, name, email, password, coins) {
        let body = {
            login: login,
            name: name,
            email : email,
            password: password,
            coins: coins
        };
        return HTTPClient.sendPost("users", body);
    }

    static addUserResponse(userId, betId, answerId, coins) {
        let body = {
            userId: userId,
            betId: betId,
            answerId: answerId,
            coins: coins
        };
        return HTTPClient.sendPost("users/answer", body);
    }

    static async dailyRewardByUserId(userId, coins) {
        await this.addCoinsByUserId(userId, coins);
        return await this.updateUserById(userId, "reward", "1");
    }

    static addImagetoUser(userId, img) {
        return HTTPClient.uploadImage("/images/upload/" + userId, img);
    }

    // PUT

    static updateUserById(id, columns, value) {
        let columnsMap = new Map();

        if (Array.isArray(columns) && Array.isArray(value))
            for (let i = 0; i < columns.length - 1; i++)
                columnsMap.set(columns[i], value[i]);
        else
            columnsMap.set(columns, value);
        return HTTPClient.sendPut("users/" + id, Common.parseColumns(columnsMap));
    }

    static updateUsersByFilters(filters, columns) {
        return HTTPClient.sendPut(Common.parseFilters("users", filters), Common.parseColumns(columns));
    }

    static updateAllUsers(columns, value) {
        if (arguments.length === 2) {
            let columnsMap = new Map();
            columnsMap.set(columns, value);
            columns = columnsMap;
        }
        return this.updateUsersByFilters([], columns);
    }

    static addCoinsByUserId(userId, coins) {
        let body = {
            "coins": coins
        };
        return HTTPClient.sendPut("users/" + userId + "/addCoins", body);
    }

    static addCoinsByFilters(filters, coins) {
        let body = {
            "coins": coins
        };
        return HTTPClient.sendPut(Common.parseFilters("users/addCoins", filters), body);
    }

    // DELETE

    static deleteUserById(id) {
        return HTTPClient.sendDelete("users/" + id);
    }

    static deleteUserByFilters(filters) {
        return HTTPClient.sendDelete(Common.parseFilters("users", filters));
    }

    static resetUserResponses(filters) {
        return HTTPClient.sendDelete(Common.parseFilters("users/answer", filters));
    }

    // GET

    static getUserById(id) {
        return HTTPClient.sendGet("users/" + id);
    }

    static getUsersByFilters(filters) {
        return HTTPClient.sendGet(Common.parseFilters("users", filters));
    }

    static getAllUsers() {
        return this.getUsersByFilters([]);
    }

    static getUsersByName(name) {
        let filters = new Map();
        filters.set("name", name);
        return this.getUsersByFilters(filters);
    }

    static getUsersByLogin(login) {
        let filters = new Map();
        filters.set("login", login);
        return this.getUsersByFilters(filters);
    }

    static getUserByEmail(email) {
        let filters = new Map();
        filters.set("email", email);
        return this.getUsersByFilters(filters);
    }
}

module.exports = UsersService;
