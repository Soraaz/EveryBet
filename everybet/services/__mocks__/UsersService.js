"use strict";

class UsersService {
    static addUserResponse = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        resolve();
    }));

    static removeCoinsByUserId = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        resolve();
    }));

    static addCoinsByUserId = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        resolve();
    }));

    static getUserById = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        let res = [{
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": "images/uploads/b77036a7cb9344dfc7b901e07097e036.png",
            "coins": 999999999,
            "country": null,
            "email": "a.a@a.fr",
            "expoPushToken": "",
            "id": 15,
            "login": "Altan",
            "name": "",
            "password": "43102984a36d9196dd2d92cf10de63cd44226bcc6f8e5135a9780a9c8303a44fb687ba6134b4d4b68259aac57d7ff0f2ab6ca877019b75752dda8cf03a1c18ac",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }];
        resolve(res);
    }));

    static getUserByEmail = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        let res = [{
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": null,
            "coins": 110,
            "country": null,
            "email": "antoine.dury@epitech.eu",
            "expoPushToken": "",
            "id": 3,
            "login": "Anto",
            "name": "Antoine",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }];
        resolve(res);
    }));

    static getAllUsers = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        let res = [{
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": "images/uploads/ed8d8b46fbe0747ccf5735e0b88bfb2d.png",
            "coins": 11111,
            "country": null,
            "email": "Sora333@hotmail.fr",
            "expoPushToken": "",
            "id": 20,
            "login": "Sora2",
            "name": "",
            "password": "267f9f0b7fb7ad7209367f9b37db600c51b4729ca2eb842a7c125e107fd2531c36e19cbdc0cfee35d15ef927a1a222fab97742546cd973c69b145781036a080d",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": null,
            "coins": 10000,
            "country": null,
            "email": "Jaja@gmail.com",
            "expoPushToken": "",
            "id": 18,
            "login": "Jaja",
            "name": "Ja",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": "images/uploads/456a8be835912a7f1e195607be360274.png",
            "coins": 509,
            "country": null,
            "email": "Sora322@hotmao.fr",
            "expoPushToken": "ExponentPushToken[5tFwMaOMWh4K4CTfpJzgRR]",
            "id": 17,
            "login": "Sora3",
            "name": "Dhdb",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 1,
            "avatar": "images/uploads/4c72d8ac63711879718a5b87055f41dd.png",
            "coins": 131,
            "country": null,
            "email": "charleslebastard@hotmail.fr",
            "expoPushToken": "",
            "id": 4,
            "login": "Nalmanak",
            "name": "Charles",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": null,
            "coins": 110,
            "country": null,
            "email": "antoine.dury@epitech.eu",
            "expoPushToken": "",
            "id": 3,
            "login": "Anto",
            "name": "Antoine",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": null,
            "coins": 110,
            "country": null,
            "email": "Momo@gmail.com",
            "expoPushToken": "",
            "id": 19,
            "login": "Momo",
            "name": "Momo",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": "images/uploads/b77036a7cb9344dfc7b901e07097e036.png",
            "coins": 0,
            "country": null,
            "email": "a.a@a.fr",
            "expoPushToken": "",
            "id": 15,
            "login": "Altan",
            "name": "",
            "password": "43102984a36d9196dd2d92cf10de63cd44226bcc6f8e5135a9780a9c8303a44fb687ba6134b4d4b68259aac57d7ff0f2ab6ca877019b75752dda8cf03a1c18ac",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }];
        resolve(res);
    }));
}

module.exports = UsersService;
