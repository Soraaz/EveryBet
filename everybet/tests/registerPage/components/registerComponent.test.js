import React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme/build";
import UsersService from '../../../services/UsersService';

import RegisterComponent, {
	login,
	email,
	password,
	passwordConfirm,
	terms
} from "../../../src/registerPage/components/registerComponent.js";

describe("RegisterComponent", () => {
	describe("rendering", () => {
		it("Should render correctly", () => {
			let registerCallbackSpy = sinon.spy();
			const tree = renderer
				.create(<RegisterComponent callback={registerCallbackSpy}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () => {
		let component;
		let cb = sinon.spy();

		beforeEach(() => {
			component = new RegisterComponent();
		});

		it("Should instantiate the class", () => {
			expect(component).to.be.an.instanceOf(RegisterComponent);
		});

		it("Should check the login", () => {
			expect(login("test")).to.be.equal("test");
		});

		it("Should check the email", () => {
			expect(email("test@test.fr")).to.be.equal("test@test.fr");
		});

		it("Should check the password", () => {
			expect(password("testtest")).to.be.equal("testtest");
		});

		it("Should check the password Confirm", () => {
			expect(passwordConfirm("testtest")).to.be.equal("testtest");
		});

		it("Should check the terms", () => {
			expect(terms(true)).to.be.equal(true);
		});

		describe("validation steps", () => {
			let wrapper;
			let instance;

			beforeEach(() => {
				wrapper = shallow(<RegisterComponent callback={cb}/>);
				expect(cb).to.be.not.called;
				instance = wrapper.instance();
			});

			it("Should check if the login already exists", (done) => {
				UsersService.getUsersByLogin = jest.fn().mockReturnValue(new Promise(async (resolve) => {
					let res = [{
						"test": "test",
					}];
					resolve(res);
				}));
				const spy = sinon.spy(instance, "_loginExists");

				expect(spy).to.be.not.called;
				instance._checkLogin("this_user_never_exist");
				setTimeout(() => {
					expect(spy).to.be.calledOnce;
					done();
				}, 200);
			});

			it("Should check if the login doesn't exist", (done) => {
				UsersService.getUsersByLogin = jest.fn().mockReturnValue(new Promise(async (resolve) => {
					let res = [];
					resolve(res);
				}));
				const spy = sinon.spy(instance, "_loginDoesNotExist");

				expect(spy).to.be.not.called;
				instance._checkLogin("this_user_never_exist");
				setTimeout(() => {
					expect(spy).to.be.calledOnce;
					done();
				}, 200);
			});
		});
	});
});
