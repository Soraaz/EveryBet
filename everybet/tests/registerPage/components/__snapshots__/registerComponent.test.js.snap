// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterComponent rendering Should render correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
      "marginTop": 20,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Identifiant"
          placeholderTextColor="#DBDBDB"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 3,
              "fontFamily": "System",
              "fontSize": 15,
              "height": 50,
              "marginTop": 20,
              "paddingLeft": 20,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 3,
              "width": 282,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="E-mail"
          placeholderTextColor="#DBDBDB"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 3,
              "fontFamily": "System",
              "fontSize": 15,
              "height": 50,
              "marginTop": 20,
              "paddingLeft": 20,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 3,
              "width": 282,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mot de passe"
          placeholderTextColor="#DBDBDB"
          secureTextEntry={true}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 3,
              "fontFamily": "System",
              "fontSize": 15,
              "height": 50,
              "marginTop": 20,
              "paddingLeft": 20,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 3,
              "width": 282,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Confirmation du mot de passe"
          placeholderTextColor="#DBDBDB"
          secureTextEntry={true}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 3,
              "fontFamily": "System",
              "fontSize": 15,
              "height": 50,
              "marginTop": 20,
              "paddingLeft": 20,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 3,
              "width": 282,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginRight": 30,
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "stretch",
              "flex": 4,
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          J'accepte les conditions d'utilisation
        </Text>
        <RCTSwitch
          accessibilityLabel={null}
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "height": 31,
                "width": 51,
              },
              Array [
                Object {
                  "alignSelf": "flex-end",
                },
                Object {
                  "backgroundColor": "#989898",
                  "borderRadius": 16,
                },
              ],
            ]
          }
          thumbTintColor="#F1F1F1"
          tintColor="#989898"
          value={false}
        />
      </View>
    </View>
  </View>
  <Text
    onPress={[Function]}
  >
    Link to Terms
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#A6E48D",
        "borderRadius": 10,
        "height": 45,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.33,
        "shadowRadius": 7.5,
        "width": 215,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      S'inscrire
    </Text>
  </View>
</View>
`;
